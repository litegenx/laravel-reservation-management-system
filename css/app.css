.router-link-active {
  cursor: pointer;
}

