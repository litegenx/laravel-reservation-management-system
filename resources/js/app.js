require('./bootstrap');
import 'babel-polyfill';
import Vue from 'vue';
import router from './router';
import store from './store';
import App from './components/App';
import vuetify from './plugins/vuetify';

new Vue({
    el: '#app',
    store,
    router,
    components: { App },
    template: '<App/>',
    vuetify,
});
