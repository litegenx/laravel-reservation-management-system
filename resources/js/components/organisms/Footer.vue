<template>
    <v-footer
        app
        dark
        color="primary"
        padless
        absolute
    >
        <v-layout
            justify-center
            wrap
        >
            <v-btn
                v-for="item in sns"
                :key="item.icon"
                class="mx-4 white--text"
                icon
                :href="item.url"
                target="_blank"
            >
                <v-icon size="24px">
                    {{ item.icon }}
                </v-icon>
            </v-btn>
            <v-flex
                secondary
                lighten-2
                py-4
                text-center
                white--text
                xs12
            >
                {{ year }} â€” <strong>{{ $t('title') }}</strong>
            </v-flex>
        </v-layout>
    </v-footer>
</template>

<script>
    import { mapGetters } from 'vuex';

    export default {
        computed: {
            ...mapGetters({
                sns: 'getSns',
            }),
            year () {
                return new Date().getFullYear();
            },
        },
    };
</script>
