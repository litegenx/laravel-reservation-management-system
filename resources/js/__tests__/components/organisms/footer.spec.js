import { mount } from '@vue/test-utils';
import Vuex from 'vuex';
import Footer from '../../../components/organisms/Footer';
import app from '../app';
import setupLocalVue from '../setupLocalVue';

describe('Footer', () => {
    it('should render footer', () => {
        const wrapper = mount(app(Footer, 'bottom'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    getSns: () => [
                        { icon: 'mdi-github-circle', url: 'https://github.com' },
                        { icon: 'mdi-twitter-circle', url: 'https://twitter.com' },
                    ],
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot();

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-icon')).toHaveLength(2);
        expect(wrapper.find('strong').text()).toBe('Reservation System');
        expect(wrapper.find('div.flex').text()).toContain(new Date().getFullYear());
    });
});
