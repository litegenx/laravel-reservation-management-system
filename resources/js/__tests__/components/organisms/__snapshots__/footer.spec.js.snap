// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render footer 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
    <footer
      class="v-footer v-sheet v-sheet--tile theme--dark v-footer--absolute v-footer--padless primary"
      style="left: 0px; right: 0px; bottom: 0px;"
    >
      <div
        class="layout justify-center wrap"
      >
        <a
          class="mx-4 white--text v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default"
          href="https://github.com"
          target="_blank"
        >
          <span
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-github-circle theme--dark"
              style="font-size: 24px;"
            />
          </span>
        </a>
        <a
          class="mx-4 white--text v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default"
          href="https://twitter.com"
          target="_blank"
        >
          <span
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-twitter-circle theme--dark"
              style="font-size: 24px;"
            />
          </span>
        </a>
         
        <div
          class="flex secondary lighten-2 py-4 text-center white--text xs12"
        >
          
            2019 â€” 
          <strong>
            Reservation System
          </strong>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
